molo/forms/tests/test_templatetags.py

```diff
@@ -320,12 +320,30 @@ def test_forms_list_linked_to_pages(self):
         })
         survey = self.direct_molo_form_page
         article = self.mk_article(self.main)
+
         article_page_form = ArticlePageForms(form=survey, page=article)
         article.forms.add(article_page_form)
         res = forms_list_linked_to_pages(context, article)
         self.assertEqual(len(res['forms']), 1)
         self.assertEqual(res['forms'][0]['molo_form_page'], survey)
 
+        sub_section = self.mk_sections(article, count=1)[0]
+        sub_article = self.mk_article(sub_section)
+
+        survey2, survey2_form_field = (
+            self.create_molo_form_page(
+                parent=self.forms_index,
+                title="direct form title",
+                slug="direct_form_title",
+                display_form_directly=True,
+            ))
+
+        article_page_form = ArticlePageForms(form=survey2, page=sub_article)
+        sub_article.forms.add(article_page_form)
+        res = forms_list_linked_to_pages(context, sub_article)
+        self.assertEqual(len(res['forms']), 1)
+        self.assertEqual(res['forms'][0]['molo_form_page'], survey2)
```

Comment:
After everything please call `forms_list_linked_to_pages` again for the parent article and check that only the first survey is there.